### PR TITLE
Transition to Pipenv and update python/js dependencies

### PR DESCRIPTION
**Pull request by Jkrzy**
## Description

Per #704, Transition requirements.txt to Pipfile.

Pipfile.lock included with current versions of all dependencies. Django is pinned at 1.11 until we're ready for that transition. DRF is pinned at 3.6, code changes required for >=3.7.

Dockerfile revised to install packages with Pipenv.

### CODE REVIEW

**Comment by rogeruiz** [comment]
@jkrzy are these changes going to affect the start commands for the build pack?

http://docs.cloudfoundry.org/buildpacks/python/index.html

**Comment by rogeruiz** [comment]
Also @jkrzy, looks like you’ll need to update `.circleci/config.yml` to use the new commands. Might make sense to have CI leverage a script now rather than raw shell commands.

**Comment by Jkrzy** [comment]
Thanks for the [wip] review @rogeruiz! The CF buildpack generates a ``requirements.txt`` from the Pipfile, no additional changes needed there but deploy docs do need a touch-up.

Doc and CI changes are incoming, need to familiarize myself with the deploy process before I update the documentation on it 😃 🏫

**Comment by rogeruiz** [comment]
Let me know if you need CF credentials. I can give you access to staging.

File: .circleci/config.yml [hunk 1]
```diff
@@ -57,6 +57,7 @@ jobs:
             python manage.py test --noinput --settings=tock.settings.test
             bandit -r . -x docker_entrypoint.py
             flake8
+            pipenv check --dev
```
Comment (rogeruiz) [inline .circleci/config.yml]: Great, I'll add this to the SSP. What does this check do for the codebase? Could you link me to documentation or write up a short summary of the feature here?

Comment (Jkrzy) [reply .circleci/config.yml]: Ref: https://docs.pipenv.org/advanced/#detection-of-security-vulnerabilities

Quick summary: Pipenv leverages the [Safety](https://github.com/pyupio/safety) package to check dependencies against known vulnerabilities. Vulnerabilities are sourced from [Safety-DB](https://github.com/pyupio/safety-db) which are updated once a month by https://pyup.io/.

I would consider this a complement, not a replacement, to active vulnerability scanning a-la gemnasium/snyk/etc.

**Comment by Jkrzy** [comment]
@rogeruiz, js packages updated. This is ready for review at your leisure. 📦 ⬆️ 

In reviewing on my end, I didn't come across any breaking changes between the previous->these package versions. 

Per the earlier [comment](https://github.com/18F/tock/pull/762#issuecomment-374614672), I think the changes required here for the circleci config to resolve #704 are minor enough to not trigger a more broad re-working of `config.yml`. Lets tackle that in a new issue if it is a priority, what do you think???

**Comment by rogeruiz** [comment]
@Jkrzy Looks like Gemnasium is [still saying we're out of date](https://gemnasium.com/github.com/18F/tock), but not sure why since we've updated most of these dependencies before you created this PR. 🤷‍♂️ Worth noting though: [Dependency Status](https://gemnasium.com/github.com/18f/tock)

**Comment by rogeruiz** [comment]
Also, could be related, but it looks like Snyk, which may ultimately be the Gemnasium replacement, doesn't support `Pipfile` to scan for dependencies. Is there a way to still leverage `requirements.txt` with a `Pipfile`?

🐍 [source](https://snyk.io/docs/snyk-for-python)

**Comment by Jkrzy** [comment]
Yeah, looks like Gemnasium does not support Pipfiles either. ☹️ 

Pipenv can generate `requirements.txt` from the `Pipfile.lock`, I'll follow-up with a PR to accommodate.

Pipenv having the [blessings of python.org](https://packaging.python.org/tutorials/managing-dependencies/) will hopefully drive adoption by external tools such as these.

**Pull request merged**